core/trino-main/src/main/java/io/trino/sql/planner/ConnectorExpressionTranslator.java

```diff
@@ -297,6 +335,32 @@ public SqlToConnectorExpressionTranslator(Session session, Map<NodeRef<Expressio
             return Optional.of(new FieldDereference(typeOf(node), translatedBase.get(), (int) (((LongLiteral) node.getIndex()).getValue() - 1)));
         }
 
+        @Override
+        protected Optional<ConnectorExpression> visitInPredicate(InPredicate node, Void context)
+        {
+            InListExpression valueList = (InListExpression) node.getValueList();
+            Optional<ConnectorExpression> valueExpression = process(node.getValue());
+
+            if (valueExpression.isEmpty()) {
+                return Optional.empty();
+            }
+
+            ImmutableList.Builder<ConnectorExpression> values = ImmutableList.builder();
+
+            for (Expression value : valueList.getValues()) {
+                Optional<ConnectorExpression> processedValue = process(value);
+
+                if (processedValue.isEmpty()) {
+                    return Optional.empty();
+                }
+
+                values.add(processedValue.get());
+            }
+
+            ConnectorExpression inListExpression = new Call(typeOf(node.getValueList()), StandardFunctions.IN_LIST_FUNCTION_NAME, values.build());
```

Comment:
the structure

```
in(value, list(a, b, c))
```

models what we have in the engine, but is not something we have to do in connector expressions. cc @martint

Reply:
I've introduced a list type (but I can remove it after next review round if you see it as redundant).

Reply:
PTAL now @findepi

Reply:
Modeling it as a list is the right thing. Luckily, we already have such a concept — an array type!

Reply:
then how we call the array constructor. Just `"array"` Call?

Reply:
I've translated `x IN (y...)` to `$in(x, $array(y...))` form. Will do this the trick?

Reply:
Yes, that’s good

Reply:
At some point we may want to introduce array literals to the IR and we’ll replace that call with it